tests/unittests/InterpreterTest.cpp

```diff
@@ -898,6 +898,89 @@ TEST(Interpreter, nonLinearClassifier) {
   }
 }
 
+/// Generate images in two classes. A "line" is labeled as 0
+/// and a "cross" is labeled as 1
+static void generateImageData(Tensor &images, Tensor &labels, unsigned numSamples, bool isTrain) {
+  auto L = labels.getHandle<size_t>();
+  auto image = images.getHandle<>();
+  images.zero();
+  
+  for (size_t i = 0; i < numSamples; i++) {
+    bool isLine = i % 2 == 0;
+    L.at({i, 0}) = isLine ? 0 : 1;
+    size_t target, end;
+    target = nextRandInt(2, 5);
+    // to make sure testing set is not totally included in 
+    // training set
+    end = isTrain ? 6 : 7;
+
+    for (size_t x = 0; x < end; x++) {
+      for (size_t y = 0; y < end; y++) {
+        if (isLine) {
+	  // line
+	  if (x == target)
+            image.at({i, x, y, 0u}) = 1;
+        } else {
+          // cross
+          if (x == target || y == target)
+            image.at({i, x, y, 0u}) = 1;
+        }
+      }
+    }
+  }
+}
+
+TEST(Interpreter, simpleConv) {
+  unsigned numSamples = 1000;
+  unsigned batchSize = 7;
+  ExecutionEngine EE;
+
+  EE.getConfig().learningRate = 0.01;
+  EE.getConfig().batchSize = batchSize;
+  EE.getConfig().momentum = 0.9;
+
+  auto &mod = EE.getModule();
+  Function *F = mod.createFunction("main");
+
+  auto *input = mod.createVariable(ElemKind::FloatTy, {batchSize, 8, 8, 1}, "input",
+                                   VisibilityKind::Public, Variable::TrainKind::None);
+
+  auto *ex = mod.createVariable(ElemKind::IndexTy, {batchSize, 1}, "exp",
+                                VisibilityKind::Public, Variable::TrainKind::None);
+
+  auto *CV0 = F->createConv("conv", input, 1, 3, 1, 0, 1);
+  auto *RL0 = F->createRELU("relu", CV0);
+  auto *FCL0 = F->createFullyConnected("fc", RL0, 2);
+  auto *SM = F->createSoftMax("sm", FCL0, ex);
+  auto *result = F->createSave("ret", SM);
+  Function *TF = glow::differentiate(F, EE.getConfig());
+  EE.compile(CompilationMode::Train, TF);
+
+  Tensor images(ElemKind::FloatTy, {numSamples, 8, 8, 1});
+  Tensor labels(ElemKind::IndexTy, {numSamples, 1});
+  generateImageData(images, labels, numSamples, true);
+
+  // Training:
+  EE.runBatch(500, {input, ex}, {&images, &labels});
+  EE.compile(CompilationMode::Infer, F);
+
+  // generate the images used for testing.
+  Tensor testImages(ElemKind::FloatTy, {batchSize, 8, 8, 1});
+  Tensor testLabels(ElemKind::IndexTy, {batchSize, 1});
+  generateImageData(testImages, testLabels, batchSize, false);
+  EE.run({input}, {&testImages});
+  int success = 0;
+  auto SMH = result->getVariable()->getHandle<>();
+  for (size_t i = 0; i < batchSize; i++) {
+    bool isLine = (i % 2 == 0);
+    auto A = SMH.at({i, 0});
+    auto B = SMH.at({i, 1});
+    if ((isLine && A > B) || (!isLine && B > A))
+      success++;
+  }
+  EXPECT_GT((float)success / batchSize, 0.99);
```

Comment:
nit: We can simply check that success == batchSize (success needs to be changed to unsigned in this case, to avoid signed to unsigned comparison)